File: qc-application/src/main/java/com/eager/questioncloud/application/api/feed/FeedSubscribeService.java
package com.eager.questioncloud.application.api.feed;

import com.eager.questioncloud.core.common.PagingInformation;
import com.eager.questioncloud.core.domain.creator.dto.CreatorInformation;
import com.eager.questioncloud.core.domain.subscribe.infrastructure.SubscribeRepository;
import java.util.List;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class FeedSubscribeService {
    private final SubscribeRepository subscribeRepository;

    public List<CreatorInformation> getMySubscribeCreators(Long userId, PagingInformation pagingInformation) {
        return subscribeRepository.getMySubscribeCreators(userId, pagingInformation);
    }

    public int countMySubscribe(Long userId) {
        return subscribeRepository.countMySubscribe(userId);
    }
}

File: qc-application/src/main/java/com/eager/questioncloud/application/business/subscribe/service/FeedSubscribeService.kt
package com.eager.questioncloud.application.business.subscribe.service

import com.eager.questioncloud.core.common.PagingInformation
import com.eager.questioncloud.core.domain.creator.dto.CreatorInformation
import com.eager.questioncloud.core.domain.creator.infrastructure.repository.CreatorRepository
import com.eager.questioncloud.core.domain.subscribe.infrastructure.repository.SubscribeRepository
import org.springframework.stereotype.Service

@Service
class FeedSubscribeService(
    private val subscribeRepository: SubscribeRepository,
    private val creatorRepository: CreatorRepository,
) {
    fun getMySubscribes(userId: Long, pagingInformation: PagingInformation): List<CreatorInformation> {
        val subscribedCreatorIds = subscribeRepository.getMySubscribedCreators(userId, pagingInformation)
        return creatorRepository.getCreatorInformation(subscribedCreatorIds)
    }

    fun countMySubscribe(userId: Long): Int {
        return subscribeRepository.countMySubscribe(userId)
    }
}
